File: app/src/main/java/com/bmn/bookfinder/data/room/DatabaseAsync.java
package com.bmn.bookfinder.data.room;

import android.os.AsyncTask;

import java.lang.ref.WeakReference;
import java.util.ArrayList;

/**
 * Created by Ndriçim Sadiku on 30 June 2020
 * ndricim@frakton.com
 */
public class DatabaseAsync extends AsyncTask<ArrayList<BookEntity>, Long, Boolean> {

    private BookDao bookDao;
    private Listener listener;

    public DatabaseAsync(BookDao dao, Listener listener) {
        this.bookDao = dao;
        this.listener = listener;
    }

    @Override
    protected Boolean doInBackground(ArrayList<BookEntity>... arrayLists) {
        bookDao.insertBooks(arrayLists[0]);
        return true;
    }

    @Override
    protected void onPostExecute(Boolean aBoolean) {
        super.onPostExecute(aBoolean);
        listener.onDone(aBoolean);
    }

   public interface Listener {
        void onDone(boolean success);
    }
}

File: app/src/main/java/com/bmn/bookfinder/data/room/DatabaseAsync.kt
package com.bmn.bookfinder.data.room

import android.os.AsyncTask
import java.util.*

/**
 * Created by Ndriçim Sadiku on 30 June 2020
 * ndricim@frakton.com
 */
class DatabaseAsync(private val bookDao: BookDao, private val listener: (Boolean) -> Unit) :
    AsyncTask<ArrayList<BookEntity?>?, Long?, Boolean>() {
    override fun doInBackground(vararg params: ArrayList<BookEntity?>?): Boolean {
        bookDao.insertBooks(params[0])
        return true
    }

    override fun onPostExecute(aBoolean: Boolean) {
        super.onPostExecute(aBoolean)
        listener(aBoolean)
    }
}